### ISSUE TITLE
Suggest use `to_string` rather than `format!("{}")`

### ISSUE THREAD

**Issue by leiysky** [rust-lang/rust-clippy#8828]
### What it does

Sometime, people are using `format!("{}")` to stringify an object, which can be rewritten in `.to_string()` to keep code clean.

### Category

style

### Example

```rust
let a = format!("{}", x);
```

Could be written as:

```rust
let a = x.to_string();
```

**Comment by smoelius** [comment]
Could it be this already exists? https://rust-lang.github.io/rust-clippy/master/#useless_format

**Comment by andylokandy** [comment]
[`useless_format`](https://rust-lang.github.io/rust-clippy/master/#useless_format) only take effect if `foo` is a string, but it should be any type that implemented `Display`.

**Comment by smoelius** [comment]
LOL #697

**Comment by DanielEScherzer** [comment]
What about adding anything implementing `Display` (or even just `ToString`) as a restriction category lint, so that people *can* enable this but don't by default?

I just encountered this exact issue and wanted to have clippy flag the useless formatting
